codegen/type_converter.go

```diff
@@ -469,14 +469,27 @@ func (c *TypeConverter) genConverterForMap(
 			)
 		}
 
+		toFieldKeyID := keyID
+
+		toTypeKeyName := keyType
+		fromTypeKeyName, err := c.getGoTypeName(fromFieldType.KeySpec)
+		if err != nil {
+			return err
+		}
+		if fromTypeKeyName != toTypeKeyName {
```

Comment:
If the from-type is not convertible to the to-type this will result in a compile error in the generated code. For example, if from-type is `[]byte` and to-type is `int`. I'm not sure if there is an easy way to solve this but maybe you can think about it as a potential issue. Ideally we would point the user directly to the Thrift line that is the issue rather than in the generated code.

Reply:
that's correct, ideally key type should be exactly the same, theoretically user can set any hash-able type as key. This will fail at build time.

Raising issues for thrift files is unfortunately not an option for my use case: asking users to use `client.UUID` instead of endpoint `UUID` 3 layers down in a map struct inside endpoint thrift schema ...

On production the key types are 95% string, and 5% typedef string, we could limit this with downcast-ed basetype compare?

Reply:
That is fine with me.

Reply:
so in this case specifically, map key is restricted to strings / typedef strings here:
https://github.com/uber/zanzibar/pull/394/files#diff-b01b2823b821664c3242667b138861d1R405

unconvertible case can never happen

if we remove this restriction in the future we can add 

```
t1 := reflect.TypeOf(mapkeytype1)
t2 := reflect.TypeOf(mapkeytyp2)
return t1.ConvertibleTo(t2) && t2.ConvertibleTo(t1)
```